test/test-custom-transfers.sh

```diff
@@ -192,3 +192,88 @@ begin_test "custom-transfer-standalone"
   [ "$(echo "$objectlist" | wc -l)" -eq 12 ]
 )
 end_test
+
+begin_test "custom-transfer-standalone-urlmatch"
+(
+  set -e
+
+  # setup a git repo to be used as a local repo, not remote
+  reponame="test-custom-transfer-standalone-urlmatch"
+  setup_remote_repo "$reponame"
+
+  # clone directly, not through lfstest-gitserver
+  clone_repo_url "$REMOTEDIR/$reponame.git" $reponame
+
+  # set up custom transfer adapter to use a specific transfer agent, using a URL prefix match
+  git config lfs.customtransfer.testcustom.path lfstest-standalonecustomadapter
+  git config lfs.customtransfer.testcustom.concurrent false
+  git config remote.origin.lfsurl https://git.example.com/example/path/to/repo
+  git config lfs.https://git.example.com/example/path/.standalonetransfer testcustom
```

Comment:
It may make sense to add a nonsense `lfs.standalonetransfer` value to ensure that LFS is preferring the specific `lfs.https://git.example.com/example/path/.standalonetransfer` key.